src/app/users/users.component.ts

```diff
@@ -157,27 +157,17 @@ export class UsersComponent implements OnInit, AfterViewInit {
     const myUserId = '_users/org.couchdb.user:' + this.userService.get().name;
     this.couchService.allDocs('shelf').subscribe(shelfUser => {
       shelfUser.map(shelf => {
-        const myTeamIds = [ ...shelf.myTeamIds ] || [];
-        if(myTeamIds.indexOf(idShouldBeRemove)) {

```

Comment:
You need to check if Id exists before you update else it will update shelf of all users increasing request to couchdb